resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/kubernetes/submit/ClientV2Suite.scala

```diff
@@ -169,30 +189,85 @@ class ClientV2Suite extends SparkFunSuite with BeforeAndAfter {
             .endMetadata()
         }
       })
-    when(initContainerComponentsProvider.provideContainerLocalizedFilesResolver())
+    when(initContainerComponentsProvider.provideContainerLocalizedFilesResolver(any[String]))
       .thenReturn(containerLocalizedFilesResolver)
+    when(initContainerComponentsProvider.provideDriverPodFileMounter())
+        .thenReturn(fileMounter)
     when(submittedDependenciesSecretBuilder.build())
       .thenReturn(INIT_CONTAINER_SECRET)
     when(kubernetesClient.pods()).thenReturn(podOps)
     when(podOps.create(any())).thenAnswer(new Answer[Pod] {
       override def answer(invocation: InvocationOnMock): Pod = {
         new PodBuilder(invocation.getArgumentAt(0, classOf[Pod]))
           .editMetadata()
-          .withUid(DRIVER_POD_UID)
+            .withUid(DRIVER_POD_UID)
           .endMetadata()
-          .withKind(DRIVER_POD_KIND)
+            .withKind(DRIVER_POD_KIND)
           .withApiVersion(DRIVER_POD_API_VERSION)
           .build()
       }
     })
     when(podOps.withName(s"$APP_RESOURCE_PREFIX-driver")).thenReturn(namedPodResource)
+    when(pythonSubmissionResources.sparkJars).thenReturn(Seq.empty[String])
+    when(pythonSubmissionResources.primarySparkResource(any()))
+      .thenReturn(RESOLVED_PYSPARK_PRIMARY_FILE)
+    when(pythonSubmissionResources.pySparkFiles).thenReturn(PYSPARK_FILES.toArray)
+    when(pythonSubmissionResources.arguments).thenReturn(Array(PYSPARK_FILES.mkString(","), "500"))
+    when(pythonSubmissionResources.driverPod(
+      any[DriverInitContainerComponentsProvider],
+      mockitoEq(RESOLVED_PYSPARK_PRIMARY_FILE),
+      mockitoEq(RESOLVED_PYSPARK_FILES.mkString(",")),
+      any[String],
+      any[PodBuilder])).thenAnswer( new Answer[Pod] {
+        override def answer(invocation: InvocationOnMock) : Pod = {
+          invocation.getArgumentAt(0, classOf[DriverInitContainerComponentsProvider])
+           .provideDriverPodFileMounter().addPySparkFiles(
+            invocation.getArgumentAt(1, classOf[String]),
+            invocation.getArgumentAt(2, classOf[String]),
+            invocation.getArgumentAt(3, classOf[String]),
+            invocation.getArgumentAt(4, classOf[PodBuilder])
+          ).build()
+        }
+      })
+    when(fileMounter.addPySparkFiles(
+      mockitoEq(RESOLVED_PYSPARK_PRIMARY_FILE),
+      mockitoEq(RESOLVED_PYSPARK_FILES.mkString(",")),
+      any[String],
+      any())).thenAnswer( new Answer[PodBuilder] {
+        override def answer(invocation: InvocationOnMock) : PodBuilder = {
+          invocation.getArgumentAt(3, classOf[PodBuilder])
+          .editSpec()
+            .editMatchingContainer(new ContainerNameEqualityPredicate(
```

Comment:
That should be tested in the file mounter unit test. The Client class doesn't decide the environment variables to set - it trusts the submodule is providing the right configurations.